app/src/main/java/com/vultisig/wallet/ui/models/cosmosstaking/CosmosRedelegateViewModel.kt

```
@@ -0,0 +1,295 @@
+        viewModelScope.safeLaunch(
+            onError = { e -> setError(e.message ?: "Failed to build redelegate transaction") }
+        ) {
+            val coin = coin ?: return@safeLaunch setError("Wallet not loaded yet")
+            val vault =
+                withContext(Dispatchers.IO) { vaultRepository.get(route.vaultId) }
+                    ?: return@safeLaunch setError("Vault not found: ${route.vaultId}")
+
+            try {
+                ValidatorBech32Preflight.validate(route.validatorSrcAddress, coin.chain)
+                ValidatorBech32Preflight.validate(dstValidator.operatorAddress, coin.chain)
+            } catch (e: ValidatorBech32Preflight.ValidatorBech32Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)